File: workflow/scripts/get_pathway_barplot.R
log <- file(snakemake@log[[1]], open = "wt")
sink(log)
sink(log, type = "message")

library(tidyverse)
library(ggplot2)


pathway_file <- read.csv(snakemake@input[["pathway_file"]],
    sep = "\t", header = TRUE)
fil_pathway <- pathway_file %>% filter(False.Discovery.Rate.global.p.values <= 0.05)
pdf(file = snakemake@output[[1]], height = 10, width = 10)
ggplot(fil_pathway, aes(reorder(Name, observed.total.perturbation.accumulation),
    observed.total.perturbation.accumulation)) +
    geom_col(aes(fill = Status)) + coord_flip() +
    labs(x = "Pathway",
        y = "the observed total perturbation accumulation in the pathway",
            title = "pathways tA from SPIA of pGfdr<05") +
    theme_minimal()
dev.off()

File: workflow/scripts/get_pathway_barplot.py
import altair as alt
import pandas as pd

sys.stderr = open(snakemake.log[0], "w")

pathway_file = pd.read_csv(snakemake.input["pathway_file"], sep="\t")

barplot = alt.Chart(pathway_file).mark_bar().encode(
    x=alt.X("observed total perturbation accumulation:Q"),
    y=alt.Y("Name", sort='-x'), color='Status:N').properties(
        width=600, title = "pathways observed total perturbation accumulation from SPIA of False Discovery Rate global p-values <=0.05" ).transform_filter(
    alt.FieldLTEPredicate(field='False Discovery Rate global p-values', lte = 0.05)
)

barplot.save(snakemake.output[0])
